v0/destinations/braze/transform.js

```diff
@@ -374,7 +376,224 @@ function process(event) {
+//       // batchComplete = true;
+//     }
+//   } else {
+//     // https://developers.amplitude.com/docs/batch-event-upload-api#feature-comparison-between-httpapi-2httpapi--batch
+//     if (
+//       batchEventArray.length < AMBatchEventLimit &&
+//       JSON.stringify(batchPayloadJSON).length +
+//         JSON.stringify(incomingMessageEvent).length <
+//         AMBatchSizeLimit
+//     ) {
+//       batchEventArray.push(incomingMessageEvent); // set value
+//       batchEventJobs.push(metadata);
+//       set(
+//         batchEventResponse,
+//         "batchedRequest.body.JSON.events",
+//         batchEventArray
+//       );
+//       set(batchEventResponse, "metadata", batchEventJobs);
+//     } else {
+//       // event could not be pushed
+//       // it will be pushed again by a call from the caller of this method
+//       batchComplete = true;
+//     }
+//   }
+//   return batchComplete;
+// }
+
+function formatBatchResponse(batchPayload, metadataList, destination) {
+  const response = defaultBatchRequestConfig();
+  response.batchedRequest = batchPayload;
+  response.metadata = metadataList;
+  response.destination = destination;
+  return response;
+}
+
+function batch(destEvents) {
+  const respList = [];
+  let trackEndpoint;
+  let jsonBody;
+  let endPoint;
+  let type;
+  let counter = 0;
+  let attributesBatch = [];
+  let eventsBatch = [];
+  let purchasesBatch = [];
+  let metadataBatch = [];
+
+  for (let index = 0; index < destEvents.length; index += 1) {
+    // take out a single event
+    const ev = destEvents[index];
+    const { message, metadata, destination } = ev;
+
+    // get the JSON body
+    jsonBody = get(message, "body.JSON");
+
+    // get the type
+    endPoint = get(message, "endpoint");
+    type = endPoint && endPoint.includes("track") ? "track" : "identify";
+
+    // if it is a track keep on adding to the existing track list
+    // keep a count of event, attribute, purchases - 75 is the cap
+    if (type === "track") {
+      // keep the trackEndpoint for reuse later
+      if (!trackEndpoint) {
+        trackEndpoint = endPoint;
+      }
+
+      // look for events, attributes, purchases
+      const { events, attributes, purchases } = jsonBody;
+
+      // add only if present
+      if (attributes) {
+        attributesBatch.push(...attributes);
+      }
+
+      if (events) {
+        eventsBatch.push(...events);
+      }
+
+      if (purchases) {
+        purchasesBatch.push(...purchases);
+      }
+
+      // set the counter to max of len(attributesBatch), len(eventsBatch), len(purchasesBatch)
+      counter = Math.max(
+        attributesBatch.length,
+        eventsBatch.length,
+        purchasesBatch.length
+      );
+
+      // keep the original metadata object. needed later to form the batch
+      metadataBatch.push(metadata);
+
+      // put the cap on 75 or look for the last index
+      if (counter >= BRAZE_MAX_REQ_COUNT || index === destEvents.length - 1) {
```

Comment:
adding the props for the present event in the loop can overflow (>75) this.
should we change?